From 5a906f2c7a848d355531e6671eb2c7ed09062cb1 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Mon, 13 Aug 2018 15:23:07 +0800
Subject: [PATCH 4416/5242] MLK-19206 VPU: Modify the action for disable
 recoder

commit  165ade60834709eca8aa879f8b71cd64e9af8a18 from
https://source.codeaurora.org/external/imx/linux-imx.git

Modify the action for disable recoder

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
(cherry picked from commit 64302c586a541f8effc1e16427f50ec490918a9d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index 913a4c4..f15e93d 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -50,7 +50,7 @@
 #include "insert_startcode.h"
 
 unsigned int vpu_dbg_level_decoder = 1;
-static unsigned int vpu_frm_depth = 100;
+static int vpu_frm_depth = 100;
 
 /* Generic End of content startcodes to differentiate from those naturally in the stream/file */
 #define EOS_GENERIC_HEVC 0x7c010000
@@ -1301,9 +1301,15 @@ static void transfer_buffer_to_firmware(struct vpu_ctx *ctx, void *input_buffer,
 	if (ctx->b_dis_reorder) {
 		/* set the shared memory space control with this */
 		MediaIPFW_Video_CodecParams *pCodecPara;
+
 		add_scode(ctx, 0, BUFFLUSH_PADDING_TYPE);
 		pCodecPara = (MediaIPFW_Video_CodecParams *)ctx->dev->shared_mem.codec_mem_vir;
 		pCodecPara[ctx->str_index].uDispImm = 1;
+	} else {
+		MediaIPFW_Video_CodecParams *pCodecPara;
+
+		pCodecPara = (MediaIPFW_Video_CodecParams *)ctx->dev->shared_mem.codec_mem_vir;
+		pCodecPara[ctx->str_index].uDispImm = 0;
 	}
 
 	/*initialize frame count*/
-- 
1.7.9.5

